### PR TITLE
move package_snapshots to AccountsPackagePre ctors

### LINKED ISSUE #18815: Move snapshot package TempDir creation & hardlinking into ctor for AccountsPackagePre
snapshot_utils::package_xxx_snapshot() and do_package_snapshot() don't do much besides calling the AccountsPackagePre ctor. So, move that code into ctors for AccountsPackagePre. (Creating the TempDir and doing the hardlinking.)

_Original comment from original PR_

Heh, looking at it again, I think `package_incremental_snapshot` and  `package_snapshot` may just be better off as constructors for `AccountsPackagePre`, where logic like https://github.com/solana-labs/solana/pull/18504/files#diff-c957621ae265a12e03f7b9c5656e7aa74137223e603cc3cfcc2828dda1af5479R398-R405 could be moved into a constructor.

Also has the added benefit of offloading from the `snapshot_utils` module as it's growing rather large.

_Originally posted by @carllin in https://github.com/solana-labs/solana/pull/18504#discussion_r673756619_

### PR DESCRIPTION
**Pull request by snOm3ad**
#### Problem
The functions `snapshot_utils::package_xxx_snapshot` have minimal functionality that could
easily moved into constructors inside `snapshot_package::AccountsPackagePre` thereby 
offloading the `snapshot_utils.rs` file which is already pretty big.

More info [here](https://github.com/solana-labs/solana/pull/18504#discussion_r673756619).

#### Summary of Changes
I had to make the snapshot prefix constants inside `snapshot_utils.rs` public at the crate level 
in order to make this work. I'm not sure whether or not introducing this dependency is entirely 
good, either way the `snapshot_utils.rs` file needs a lot of rework so things will move around, 
I believe this does the work in the meantime. Any feedback will be greatly appreciated.

Fixes #18815

### CODE REVIEW

**Review by brooksprumo — CHANGES_REQUESTED** [review]
Overall looks good! Thanks for picking up this issue. After you make the changes, request a re-review from me.

File: runtime/src/snapshot_package.rs [hunk 1]
```diff
@@ -60,6 +68,144 @@ impl AccountsPackagePre {
             cluster_type,
         }
     }
+
+    /// Create a snapshot package
+    #[allow(clippy::too_many_arguments)]
+    fn new_snapshot<P>(
```
Comment (brooksprumo) [inline runtime/src/snapshot_package.rs:74]: Looks like this was missed. Please make this one new_snapshot_package.

**Review by brooksprumo — CHANGES_REQUESTED** [review]
I think just one small change.

**Comment by brooksprumo** [comment]
@carllin I reviewed this and it looks good to me. Since this issue and PR is originally from [your comment here](https://github.com/solana-labs/solana/pull/18504#discussion_r673756619), can you double check to make sure it meets your expectations?

Thanks!

**Comment by carllin** [comment]
Looks great, thanks guys for taking this! 😃

**Review by carllin — APPROVED** [review]

**Review by brooksprumo — APPROVED** [review]

**Comment by brooksprumo** [comment]
Thanks for your work here, @snOm3ad!

**Comment by snOm3ad** [comment]
> Thanks for your work here, @snOm3ad!

No problem👍I hope to contribute more in the future!

**Pull request merged**